gufe/smallmoleculecomponent.py

```diff
@@ -124,11 +125,15 @@ def __eq__(self, other):
         return hash(self) == hash(other)
 
     def to_dict(self) -> dict:
-        raise NotImplementedError()
+        """Serialize to dict representation"""
+        d = self.to_openff().to_dict()
+        return d
```

Comment:
One thing I'm not sure of is maybe we should tag this dict with a version tag, and then the from_dict keeps track... so

```python
# in to_dict
d = self.to_openff().to_dict()
d['gufe_version'] = 1  # i.e. our first schema, if/when we update our to_dict behaviour, increment this number

# in from_dict
if d['gufe_version'] == 1:
    # current stuff
elif d['gufe_version'] == 2:
   # future stuff, etc
```

Alternatively we enjoy our project youth and allow this to break horribly as its not in production yet.

Reply:
So whilst we're not close to API stable, I don't think adding functionality that takes advantage of versioning is worth it. That being said, it might be good to have it there early just purely for our sanity.